src/lib_json/json_reader.cpp

```diff
@@ -1096,8 +1090,8 @@ bool OurReader::parse(const char* beginDoc,
 }
 
 bool OurReader::readValue() {
-  if (stackDepth_ >= features_.stackLimit_) throwRuntimeError("Exceeded stackLimit in readValue().");
-  ++stackDepth_;
+  //  To preserve the old behaviour we cast size_t to int.
+  if (static_cast<int>(nodes_.size()) > features_.stackLimit_) throwRuntimeError("Exceeded stackLimit in readValue().");
```

Comment:
@cdunn2001 it have no sense to keep stackDepth_ in OurReader. New solution is use less additional memory (it's one member but it could be important on embedded devices) and still has the same computational complexity. And the tests are OK.